examples/mini-app/src/hooks/usePlaceBet.ts

```diff
@@ -255,18 +217,23 @@ async function _extractBetIdFromReceipt(
   expectedContractAddress: Hex,
   placeBetAbi: Abi,
   publicClient: ReturnType<typeof usePublicClient>,
-): Promise<string | null> {
-  if (!publicClient) throw new Error("publicClient is undefined")
-  console.log("Waiting for receipt for", txHash)
+): Promise<bigint | null> {
+  if (!publicClient) {
+    logger.error("_extractBetIdFromReceipt: publicClient is undefined")
+    throw new Error("publicClient is undefined")
+  }
+  logger.debug("_extractBetIdFromReceipt: Waiting for receipt for", txHash)
   const receipt = await publicClient.waitForTransactionReceipt({ hash: txHash })
-  console.log("Receipt received.")
+  logger.debug("_extractBetIdFromReceipt: Receipt received.")
 
   const placeBetEventDefinition = placeBetAbi.find(
     (item) => item.type === "event" && item.name === "PlaceBet",
   )
 
   if (!placeBetEventDefinition) {
-    console.warn("PlaceBet event definition not found in ABI.")
+    logger.warn(
+      "_extractBetIdFromReceipt: PlaceBet event definition not found in ABI.",
```

Comment:
if "public client undefined" condition results in an error, should this be an error too?

Can this result in people making bets and seeing nothing, just a warning in console?

Can placeBetEventDefinition be non-nullable and defined as early as possible?